### PR TITLE
Clear InfoPanel message when closed

### LINKED ISSUE #454: Error message not always displayed in InfoPanel.
Using 2.0.0 RC, an error message is not always displayed for an error.

This is because [UpdateVisibility](https://github.com/github/VisualStudio/blob/master/src/GitHub.VisualStudio.UI/UI/Controls/InfoPanel.xaml.cs#L39) is only called when the error message changes, and if it has previously been called with the same error message and then dismissed, the changed handler won't be called again because the property hasn't changed.

### PR DESCRIPTION
**Pull request by paladique**
Clears the message when the information panel is dismissed.
Fixes #454

### CODE REVIEW

File: src/GitHub.VisualStudio.UI/UI/Controls/InfoPanel.xaml.cs [hunk 1]
```diff
@@ -45,6 +45,7 @@ static void UpdateVisibility(DependencyObject d, DependencyPropertyChangedEventA
         void Button_Click(object sender, RoutedEventArgs e)
         {
             this.Visibility = Visibility.Collapsed;
+            this.Message = string.Empty;
```
Comment (shana) [inline src/GitHub.VisualStudio.UI/UI/Controls/InfoPanel.xaml.cs]: You can actually remove the `this.Visibility = Visibility.Collapsed;` line above and just do this one.
As soon as you set the message to empty, the `UpdateVisibility` method triggers and the visibility will get set to collapsed ([on line 42](https://github.com/github/VisualStudio/pull/457/files#diff-9482f34c5d4f74fce8bf57c770dac73bL42))

Side effects ftw! :smile:

File: src/GitHub.VisualStudio.UI/UI/Controls/InfoPanel.xaml.cs [hunk 2]
```diff
@@ -45,6 +45,7 @@ static void UpdateVisibility(DependencyObject d, DependencyPropertyChangedEventA
         void Button_Click(object sender, RoutedEventArgs e)
```
Comment (paladique) [inline src/GitHub.VisualStudio.UI/UI/Controls/InfoPanel.xaml.cs]: We should probably change this to something a little more descriptive? I'm :thought_balloon: `Dismiss_Click`?

Comment (shana) [inline src/GitHub.VisualStudio.UI/UI/Controls/InfoPanel.xaml.cs]: Oh yeah, excellent idea, self-descriptive code ftw!

File: src/GitHub.VisualStudio.UI/UI/Controls/InfoPanel.xaml.cs [hunk 3]
```diff
@@ -42,9 +42,9 @@ static void UpdateVisibility(DependencyObject d, DependencyPropertyChangedEventA
             control.Visibility = string.IsNullOrEmpty(control.Message) ? Visibility.Collapsed : Visibility.Visible;
         }
 
-        void Button_Click(object sender, RoutedEventArgs e)
+        void Dismiss_Click(object sender, RoutedEventArgs e)
```
Comment (shana) [inline src/GitHub.VisualStudio.UI/UI/Controls/InfoPanel.xaml.cs]: You need to change this in the xaml as well, otherwise it won't activate.

**Comment by shana** [comment]
:heart_eyes:

**Pull request merged**